lib/models/info_user.dart

```diff
@@ -16,6 +16,8 @@ class InfoUser{
 
   InfoUser(this._uid,this._fullName,this._gender,this._birthDay,this._imageURL,this._imageName,this._email,this._phone,this._favorite,this._cart);
```

Comment:
class InfoUser {
  final String uid;
  final String fullName;
  final String gender;
  final String birthDay;
  final String imageURL;
  final String imageName;
  final String email;
  final String phone;
  final List<int> favorite;
  final List<Cart> cart;

  InfoUser({
    required this.uid,
    required this.fullName,
    required this.email,
    this.gender = 'male',
    this.birthDay = '01/01/2022',
    this.imageURL = '',
    this.imageName = '',
    this.phone = '',
    this.favorite = const [],
    this.cart = const [],
  });

  factory InfoUser.fromJson(Map<String, dynamic> json) {
    return InfoUser(
        uid: json['uid'] ?? '',
        fullName: json['fullName'] ?? '',
        email: json['email'] ?? "",
        gender: json['gender'],
        birthDay: json['birthDay'],
        imageURL: json['imageURL'],
        phone: json['phone'],
        favorite: json['favorite'] as List<int>,
        cart: Cart.fromListJson(json['cart']));
  }

  Map<String, dynamic> toJson() => {
        'uid': uid,
        'fullname': fullName,
        'email': email,
        'birthday': birthDay,
        'imageURL': imageURL,
        'imageName': imageName,
        'gender': gender,
        'phone': phone,
        'favorite': favorite,
        'cart': cart
      };

  @override
  String toString() {
    // TODO: implement toString
    return '''
    uid $uid \n
    fullName $fullName \n
    email $email \n
    gender $gender \n,
    birthDay $birthDay \n
    phone $phone \n 
    imageURL $imageURL \n
    imageName $imageName \n
    ''';
  }
}
Không cần sử dụng private field trong trường hợp này.

Reply:
anh ơi, làm sao để nhận biết được việc .Trường hợp khi nào mình nên sử dụng private cho thuộc tính, khi nào dùng public

Reply:
private dùng trong trường hơp: 
    - biến đó không cho phép cập nhập dữ liệu trực tiếp mà chỉ cho phép cập nhập thông qua các hàm nội bộ.